### PR TITLE
Fix issue #1613

### PR DESCRIPTION
**Pull request by brunoarueira**
fixes tsuru/tsuru#1613

### CODE REVIEW

**Comment by andrewsmedina** [comment]
Why not to create MongoDB a case insensitive index?

**Comment by ggarnier** [comment]
I think we should validate team and pool names just like we currently do with app names: https://github.com/tsuru/tsuru/blob/18a6895c0a336edf50d1c06d8ed0d9fdbd1fea7b/app/app.go#L973-L981

**Comment by brunoarueira** [comment]
@ggarnier Should I change to use regex then? I'll change after your answer!

**Comment by andrestc** [comment]
hey @brunoarueira, I'm sorry but i've just commited a fix to fix the pool issue by using the same validation as we do for the app name (https://github.com/tsuru/tsuru/commit/089f3488acafb54c2dc49bfe383987ddc8dc8e04)

If you want, you can just update this PR to fix the team name case, by using the validation.ValidateName, and force push this branch with the changes from master.

**Comment by brunoarueira** [comment]
👍  @andrestc, I'll do in a sec.

File: auth/team.go [hunk 1]
```diff
@@ -11,18 +11,21 @@ import (
...
+        "github.com/tsuru/tsuru/permission"
+        "github.com/tsuru/tsuru/validation"
+        "gopkg.in/mgo.v2"
+        "gopkg.in/mgo.v2/bson"
 )
 
 var (
-	ErrInvalidTeamName   = errors.New("invalid team name")
+	ErrInvalidTeamName   = errors.New(
+                "Invalid pool name, pool name should have at most 63 " +
```
Comment (andrestc) [inline auth/team.go]: s/pool/team

File: auth/team.go [hunk 2]
```diff
@@ -62,19 +65,26 @@ func (t *Team) AllowedApps() ([]string, error) {
...
 
-// CreateTeam creates a team and add users to this team.
-func CreateTeam(name string, user *User) error {
-	if user == nil {
-		return errors.New("user cannot be null")
-	}
-	name = strings.TrimSpace(name)
-	if !isTeamNameValid(name) {
+func (t *Team) validate() error {
+	name = strings.TrimSpace(t.name)
```
Comment (andrestc) [inline auth/team.go]: Why trimming here?

File: auth/team_test.go [hunk 3]
```diff
@@ -78,10 +78,12 @@ func (s *S) TestCreateTeamValidation(c *check.C) {
 		{"a", ErrInvalidTeamName},
 		{"@abc", ErrInvalidTeamName},
 		{"my team", ErrInvalidTeamName},
+		{"Abacaxi", ErrInvalidTeamName},
+		{"TEAM", ErrTeamAlreadyExists},
```
Comment (andrestc) [inline auth/team_test.go]: using `validation.ValidateName` both `Abacaxi` and `TEAM` become invalid names, so I think the error to check should be `ErrInvalidTeamName`

Comment (brunoarueira) [inline auth/team_test.go]: my bad, I'll fix 😅

File: auth/team.go [hunk 4]
```diff
@@ -11,18 +11,21 @@ import (
...
+        "github.com/tsuru/tsuru/log"
+        "github.com/tsuru/tsuru/permission"
+        "github.com/tsuru/tsuru/validation"
+        "gopkg.in/mgo.v2"
+        "gopkg.in/mgo.v2/bson"
 )
 
 var (
-	ErrInvalidTeamName   = errors.New("invalid team name")
+	ErrInvalidTeamName   = errors.New(
```
Comment (andrestc) [inline auth/team.go]: could you also wrap this message on an "ValidationError" from the tsuru error package?

**Comment by andrestc** [comment]
hey @brunoarueira, are you still working on this?

**Comment by brunoarueira** [comment]
Sorry, I've been busy in the last days. I'll fix the last review and submit again.

**Comment by brunoarueira** [comment]
@andrestc 👍

**Comment by andrestc** [comment]
LGTM, thansk @brunoarueira

**Comment by brunoarueira** [comment]
You're welcome!

**Pull request merged**